panel/widgets/chat.py

```diff
@@ -30,6 +31,14 @@
 from .input import FileInput, TextInput
 
 
+def dataframe_renderer(value):
+    layout = Column(f"{len(value)} rows was uploaded")
+    if len(value)>1000:
+        value = value.sample(1000)
+        layout.insert(-1, "The data was downsampled to 1000 rows")
+    # Todo: We need to figure out how to tell users to add "perspective" to pn.extension
+    return layout.insert(-1, Perspective(object=value, sizing_mode="stretch_width", max_height=500))
```

Comment:
Can you explain why Perspective over pn.pane.DataFrame?

Reply:
Pn.pane.DataFrame is so difficult when you and the user don't know the data uploaded.

It cannot really handle a lot of data nicely. Adding scroll bars and a max height would help.

The user cannot really explore the data. Filter, sort, pivot and Chart.

Reply:
How big is the `perspective` extension?